base/iostream.jl

```diff
@@ -190,6 +203,14 @@ end
 
 Seek a stream relative to the current position.
 
+This is equivalent to `seek(s, position(s) + offset)`, except where seeking would
+seek beyond the end of the stream, where `skip` will throw an `EOFError`.
```

Comment:
I think it might be better to leave the case of `skip` beyond the end of the file as implementation-defined behavior.
Allowing `skip` to go beyond the end of the stream to match the behavior of `IOStream` in `TranscodingStreams` is possible, but I think it would require keeping track of a bunch of additional states like zlib does in https://github.com/madler/zlib/blob/5a82f71ed1dfc0bec044d9702463dbdf84ea3b71/gzguts.h#L188-L196, basically making skip lazy and then running any requested skips on every read. https://github.com/madler/zlib/blob/5a82f71ed1dfc0bec044d9702463dbdf84ea3b71/gzread.c#L276-L281